### PR TITLE
Add client option for GRPC connection pool size

### PR DESCRIPTION
**Pull request by damemi**
Fixes https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/issues/454

### CODE REVIEW

File: exporter/collector/config.go [hunk 1]
```diff
@@ -51,6 +51,8 @@ type ClientConfig struct {
 	Endpoint string `mapstructure:"endpoint"`
 	// Only has effect if Endpoint is not ""
 	UseInsecure bool `mapstructure:"use_insecure"`
+	// GRPCPoolSize sets the size of the connection pool in the GCP client
+	GRPCPoolSize int `mapstructure:"grpcPoolSize"`
```
Comment (dashpole) [inline exporter/collector/config.go:55]:
```suggestion
	GRPCPoolSize int `mapstructure:"grpc_pool_size"`
```

We use underscores in config

**Review by dashpole — COMMENTED** [review]
can you also add an integration test in https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/main/exporter/collector/integrationtest/config/config_test.go for config parsing?

File: exporter/collector/config.go (hunk 1 above)
Comment (damemi) [reply exporter/collector/config.go]: updated

**Review by dashpole — APPROVED** [review]
lgtm after tests pass

**Pull request merged**